### PR TITLE
Make the pod sandbox timeout configurable

### PR DESCRIPTION
**Pull request by chendave**
The timeout configuration may vary with different CRI runtimes, network situation etc.

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
/kind feature

**What this PR does / why we need it**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
 pod sandbox timeout when the container runtime is defined as "remote" now is configuration.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by chendave** [comment]
/assign yujuhong 

Hi yujuhong, could you pls take a look?

**Comment by chendave** [comment]
cc @derekwaynecarr @Random-Liu could you pls take a look at this one? thanks.

File: cmd/kubeadm/app/componentconfigs/kubelet_test.go [hunk 1]
```diff
@@ -106,6 +107,7 @@ var kubeletMarshalCases = []struct {
 			port: 12345
 			rotateCertificates: true
 			runtimeRequestTimeout: 0s
+			runtimeSandBoxTimeout: 0s
```
Comment (neolit123) [inline cmd/kubeadm/app/componentconfigs/kubelet_test.go:110]: hi, these changes in cmd/kubeadm are no longer required.

as you can see kubeletMarshalCases no longer exists here:
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/componentconfigs/kubelet_test.go


/remove-area kubeadm
/remove-sig cluster-lifecycle

**Pull request closed without merging**